### ISSUE TITLE
Default values for "optional" parameters

### ISSUE THREAD

**Issue by p-bro** [potfit/potfit#44]
More a question than an issue. For our UQ framework, @swishart and I are adding a number of global parameters, that are, by all intents and purposes, optional. We'd like to set a reasonable default value, but allow changing of a setting without editing any source file. Is there any way to set such a default (to avoid having long param files with mandatory parameters)? If not, what would be the best way to do it? 

- Ad-hoc for each variable does not seem very structured.
- Adding a toggle (such as usemaxch for maxchfile) does not seem sustainable.

I'd have thought a header file with defaults would be the easiest to maintain...

Any thoughts?

(side note: g_config.dp_tol may be used uninitialised).

**Comment by dschopf** [comment]
I am not sure what to make of this. If I understand you correctly you want to add some parameters and add reasonable default values if not provided by the user. But then you also want the user to be able to change the default values? How does that make sense? Can't the user just adjust the parameters itself? Or are you thinking of some kind of "preset" configurations where all of them have some particular values?

Maybe you could provide an example for this.

**Comment by p-bro** [comment]
I was thinking about true *optional* parameters. If they are not present in the parameters file, a reasonable default is used. If the user specifies a value in the parameter file, the default value is overridden by the user.

And while looking through the param.c I found my solution:

```
  if (g_files.endpot == NULL) {
    warning("endpot is missing in %s, setting it to %s_end\n", paramfile,
            g_files.startpot);
    g_files.endpot =
        (const char*)Malloc((strlen(g_files.startpot) + 5) * sizeof(char));
    sprintf((char*)g_files.endpot, "%s_end", g_files.startpot);
}
```

Maybe rephrase the text a little to not sound as harsh - would that be the way to go?

**Comment by dschopf** [comment]
Well, that's the way "regular" optional parameters are implemented. I still cannot see any difference from what you requested. If they are not present in the parameter file we use the default value and if the user specifies them we use the provided value. Logging when they are missing is optional.

**Comment by p-bro** [comment]
Yes. I had previously looked for that case but did not see it 🙈. This is why I asked in the first place. Now I've found my solution, so everything's cool.
